packages/react/src/components/header/components/navigationLink/NavigationLink.tsx

```diff
@@ -1,14 +1,38 @@
-import React, { cloneElement, useCallback, useContext, useEffect, useRef, useState } from 'react';
+import React, { MouseEventHandler, cloneElement, useCallback, useEffect, useRef, useState } from 'react';
 
 // import base styles
 import '../../../../styles/base.css';
 
 import styles from './NavigationLink.module.scss';
-import classNames from '../../../../utils/classNames';
+import { styleBoundClassNames } from '../../../../utils/classNames';
 import { Link } from '../../../link';
 import { NavigationLinkDropdown, NavigationLinkInteraction, DropdownMenuPosition } from './navigationLinkDropdown';
-import { HeaderNavigationMenuContext } from '../headerNavigationMenu/HeaderNavigationMenuContext';
+import { useHeaderNavigationMenuContext } from '../headerNavigationMenu/HeaderNavigationMenuContext';
 import { DropdownDirection } from './types';
+import { useHeaderContext } from '../../HeaderContext';
+
+const classNames = styleBoundClassNames(styles);
+
+export type LinkProps = {
+  depth: number;
```

Comment:
@timwessman Check if this is needed as a prop to NavigationLink. It is used in dropdown but is it really needed here as a prop that a user can give?